File: arrayHash/leetcode_334/solution.py
class Solution:
    def increasingTriplet(self, nums: list[int]) -> bool:
        mini, mid = float("inf"), float("inf")

        for i in nums:
            if i <= mini:
                mini = i
            elif i <= mid:
                mid = i
            else:
                return True
        return False

File: arrayHash/leetcode_334/solution.rb
# @param {Integer[]} nums
# @return {Boolean}
def increasing_triplet(nums)
    min, mid = Float::INFINITY, Float::INFINITY

    for num in nums
        if num <= min
            min = num
        elsif num <= mid
            mid = num
        else
            return true
        end
    end

    return false
    
end